plugins/catalog-backend/src/ingestion/processors/BuiltinKindsEntityProcessor.ts

```diff
@@ -193,6 +194,29 @@ export class BuiltinKindsEntityProcessor implements CatalogProcessor {
         RELATION_PART_OF,
         RELATION_HAS_PART,
       );
+
+      // TODO: Consider building a helper around this, but it's a bit difficult
+      // with "deleting" the previous field.
+
+      // TODO: Emitting the attachment from the definition field for now. That
+      // still requires that the definition is filled with the full data and we
+      // we can do any special caching around it for now. Later we can read it
+      // form a definitionLocation field instead.
+      const key = 'definition';
+      // TODO: Also support directly reading from a location in the future, this
+      // avoid that we bload our memory with all definitions.
+      const data = Buffer.from(api.spec.definition!, 'utf8');
+      // TODO: Actually a hard question, depends either on the use case or data,
+      // but for api definitions plain should be a fine choice.
+      const contentType = 'text/plain';
+      emit(result.attachment('definition', data, contentType));
+
+      // TODO: How to migrate from "definition" to "definition?", it's a breaking change?
+      delete api.spec.definition;
+      api.spec.definitionLocation = stringifyLocationReference({
+        type: 'attachment',
+        target: key,
```

Comment:
Removing the definition field is kind of strange, but required if we want to keep it for ingestion, but also keep the API response small. But this is a potential migration strategy. We could start by duplicating the data inside the attachments and later remove the fields (after the deprecation period). But we won't get the performance benefits till then.

Another idea I would like to implement here, is allowing to use the `definitionLocation` in the spec file and copy it here into the database using the url reader.